dependencies/O2CompileFlags.cmake

```diff
@@ -0,0 +1,52 @@
+# Copyright CERN and copyright holders of ALICE O2. This software is distributed
+# under the terms of the GNU General Public License v3 (GPL Version 3), copied
+# verbatim in the file "COPYING".
+#
+# See http://alice-o2.web.cern.ch/license for full licensing information.
+#
+# In applying this license CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+
+include_guard()
+
+set(CMAKE_CXX_FLAGS_COVERAGE "-g -O2 -fprofile-arcs -ftest-coverage")
```

Comment:
It sounds a good idea to me to extract the coverage and CI stuff. But I'd still set the CXXFLAGS for RELEASE, DEBUG, RELWITHDEBINFO.

Reply:
well, the way the Release, Debug and co flags is suboptimal as well : 
- why are we defining flags for Fortran at all ?
- the CACHE FORCE for CXX_FLAGS_DEBUG is wrong : it prevents developers to play with those flags (using the -D at configure time or ccmake or the cmake gui) if they need to (unless they change the CMakeLists.txt itself, which is not the intended usage of the cache variables)
- as far as I can see (omitting the assert business which is a CI concern only), compared to the defaults, to only thing we change is the optimization level for release, but this is not immediately clear given the number of lines we have for the flags.

I would go for something simpler like : 

string(CMAKE_CXX_FLAGS_DEBUG " -O0")
etc...

Reply:
- I don't know either why we set fortran flags, probably this can go.
- Don't know why this is set as cache force, I think it is OK when the user can override it. What we should consider is that the value should be updated if it changes in the CMake and it was not explicitly set by the user.
- Btw. we merged this just as a quick fix, because there were unsatisfied library dependencies which were not spotted without the --no-undefined, and the flags were not set at all. I just copied the code from the old CMake. We were kind of waiting for you to return to discuss how to proceed.